src/r1cs/verifier.rs

```diff
@@ -94,8 +86,24 @@ impl<'a, 'b> ConstraintSystem for VerifierCS<'a, 'b> {
         self.constraints.push(lc);
     }
 
-    fn challenge_scalar(&mut self, label: &'static [u8]) -> Scalar {
-        self.transcript.challenge_scalar(label)
+    fn randomized_constraints<F>(
+        &mut self,
+        label: &'static [u8],
+        callback: F,
+    ) -> Result<(), R1CSError>
+    where
+        F: 'static + Fn(&mut Self, Scalar) -> Result<(), R1CSError>,
+    {
+        match self.deferred_constraints {
+            Some(ref mut list) => {
+                list.push((label, Box::new(callback)));
+                return Ok(());
+            }
+            None => {}
+        }
+
+        let challenge = self.transcript.challenge_scalar(label);
+        callback(self, challenge)
```

Comment:
Shouldn't this be part of the `None` arm of the match?

Reply:
Yup. Will change.

Reply:
Fixed in a22c3d0